packages/react-atlas-core/src/TextArea/TextArea.js

```diff
@@ -0,0 +1,156 @@
+import React, { cloneElement } from "react";
+import PropTypes from "prop-types";
+import { InputCore } from "../Input";
+import cx from "classnames";
+
+class TextArea extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    // Initial state
+    this.state = {
+      "value": props.value || "",
+      "remaining": props.maxLength,
+      "active": false,
+      "valid": true
+    };
+  }
+
+  _handleChange = event => {
+    event.persist();
```

Comment:
This is natural react event behavior, as per Synthetic Event documentation:
https://facebook.github.io/react/docs/events.html

There is no need to do anything at componentWillUnmount since it's just references that will be automatically removed by react at unmount.